packages/pubsub/src/subscription.js

```diff
@@ -850,114 +688,7 @@ Subscription.prototype.setAckDeadline = function(options, callback) {
- * how many message listeners are assigned, and then removed, making sure
- * polling is handled automatically.
- *
- * As long as there is one active message listener, the connection is open. As
- * soon as there are no more message listeners, the connection is closed.
- *
- * @private
- *
- * @example
- * subscription.listenForEvents_();
- */
-Subscription.prototype.listenForEvents_ = function() {
-  var self = this;
-
-  this.on('newListener', function(event) {
-    if (event === 'message') {
-      self.messageListeners++;
-      if (self.closed) {
-        self.closed = false;
-        self.startPulling_();
-      }
-    }
-  });
-
-  this.on('removeListener', function(event) {
-    if (event === 'message' && --self.messageListeners === 0) {
-      self.closed = true;
-
-      if (self.activeRequest_ && self.activeRequest_.abort) {
-        self.activeRequest_.abort();
-      }
-    }
-  });
-};
-
-/**
- * Update the status of `maxInProgress`. Å subscription becomes "paused" (not
- * pulling) when the number of messages that have yet to be ack'd or skipped
- * exceeds the user's specified `maxInProgress` value.
- *
- * This will start pulling when that event reverses: we were paused, but one or
- * more messages were just ack'd or skipped, freeing up room for more messages
- * to be consumed.
- *
- * @private
- */
-Subscription.prototype.refreshPausedStatus_ = function() {
-  var isCurrentlyPaused = this.paused;
-  var inProgress = Object.keys(this.inProgressAckIds).length;
-
-  this.paused = inProgress >= this.maxInProgress;
-
-  if (isCurrentlyPaused && !this.paused && this.messageListeners > 0) {
-    this.startPulling_();
-  }
-};
-
-/**
- * Poll the backend for new messages. This runs a loop to ping the API at the
- * provided interval from the subscription's instantiation. If one wasn't
- * provided, the default value is 10 milliseconds.
- *
- * If messages are received, they are emitted on the `message` event.
- *
- * Note: This method is automatically called once a message event handler is
- * assigned to the description.
- *
- * To stop pulling, see {module:pubsub/subscription#close}.
- *
- * @private
- *
- * @example
- * subscription.startPulling_();
- */
-Subscription.prototype.startPulling_ = function() {
-  var self = this;
-
-  if (this.closed || this.paused) {
-    return;
-  }
-
-  var maxResults;
-
-  if (this.maxInProgress < Infinity) {
-    maxResults = this.maxInProgress - Object.keys(this.inProgressAckIds).length;
-  }
-
-  this.pull({
-    returnImmediately: false,
-    maxResults: maxResults
-  }, function(err, messages, apiResponse) {
-    if (err) {
-      self.emit('error', err, apiResponse);
-    }
-
-    if (messages) {
-      messages.forEach(function(message) {
-        self.emit('message', message, apiResponse);
-      });
-    }
-
-    setTimeout(self.startPulling_.bind(self), self.interval);
-  });
+  return this.pubsub.snapshot.call(this, name);
 };
 
```

Comment:
At a glance, the histogram implementation looks like a pretty direct port; you should be fine (provided that _my_ fairly untested implementation works).